libs/core/src/lib/toolbar/toolbar.component.ts

```diff
@@ -0,0 +1,203 @@
+import {
+    Component,
+    ViewEncapsulation,
+    AfterViewInit,
+    AfterViewChecked,
+    ChangeDetectionStrategy,
+    ElementRef,
+    ViewChild,
+    OnInit,
+    forwardRef,
+    ContentChildren,
+    QueryList,
+    HostListener,
+    ChangeDetectorRef,
+    Renderer2,
+    Input
+} from '@angular/core';
+import { ToolbarItemDirective } from './public_api';
+import { applyCssClass, CssClassBuilder } from '../utils/public_api';
+
+const ELEMENT_MARGIN = 8;
+const OVERFLOW_SPACE = 32;
+
+export type ToolbarType = 'solid' | 'transparent' | 'auto' | 'info';
+
+export type ToolbarSize = 'cozy' | 'compact';
+@Component({
+    selector: 'fd-toolbar',
+    templateUrl: './toolbar.component.html',
+    styleUrls: ['./toolbar.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ToolbarComponent implements OnInit, AfterViewInit, AfterViewChecked, CssClassBuilder {
+    /** Property allows user to pass additional class
+     */
+    @Input()
+    class: string;
+
+    /** Determines if overflow button should be visible
+     * Default value: false
+     */
+    @Input()
+    shouldOverflow: boolean = false;
+
+    /** Determines the type of toolbar
+     * Available options: 'solid' | 'transparent' | 'auto' | 'info'
+     * Default value: 'solid'
+     */
+    @Input()
+    fdType: ToolbarType = 'solid';
+
+    /** Determines the size of toolbar
+     * Available options: 'cozy' | 'compact'
+     * Default value: 'compact'
+     */
+    @Input()
+    size: ToolbarSize = 'compact';
+
+    /** Determines if toolbar contains text which size is equal to h4
+     * Default value: false
+     */
+    @Input()
+    hasTitle: boolean = false;
+
+    /** @hidden */
+    @ViewChild('toolbar')
+    toolbar: ElementRef;
+
+    /** @hidden */
+    @ViewChild('overflowBody')
+    overflowBody: ElementRef;
+
+    /** @hidden */
+    @ViewChild('overflowSpacer')
+    overflowSpacer: ElementRef;
+
+    /** @hidden */
+    @ContentChildren(forwardRef(() => ToolbarItemDirective))
+    toolbarItems!: QueryList<ToolbarItemDirective>;
```

Comment:
Re: exclamation mark - I think there are certain cases when there might not be toolbar item children, for example if the toolbar just has a title

Reply:
leftover, nice catch :)

Reply:
Still there, maybe forgot to stage for commit?

Reply:
I forgot to push changes, now its removed